### PR TITLE
Homebrew update on release

### PR DESCRIPTION
**Pull request by aguillenv**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -138,3 +138,18 @@ jobs:
         tag: ${{ github.ref }}
         file: /tmp/anycloud-macos.tar.gz
         asset_name: anycloud-macos.tar.gz
+
+    - name: Update homebrew
+      # TODO: should we create a new token or reuse the AUTOPUBLISH one?
+      run: |
+        alanversion=$(echo ${{github.ref}} | sed "s/refs\/tags\///")
+        [[ $alanversion != *'beta'* ]] && \
```
Comment (dfellis) [inline .github/workflows/release.yml:146]: Is this syntax right? The stars on the outside of the single quotes is so weird.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -138,3 +138,18 @@ jobs:
         tag: ${{ github.ref }}
         file: /tmp/anycloud-macos.tar.gz
         asset_name: anycloud-macos.tar.gz
+
+    - name: Update homebrew
+      # TODO: should we create a new token or reuse the AUTOPUBLISH one?
```
Comment (dfellis) [inline .github/workflows/release.yml:143]: Reusing the autopublish token should be fine, I think.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -138,3 +138,18 @@ jobs:
...
+        alanversion=$(echo ${{github.ref}} | sed "s/refs\/tags\///")
+        [[ $alanversion != *'beta'* ]] && \
+        curl \
+        --fail \
+        -XPOST \
+        -H "Accept: application/vnd.github.everest-preview+json" \
+        -H "Content-Type: application/json" \
+        -H "authorization: token ${{secrets.AUTOPUBLISH}}" \
+        https://api.github.com/repos/alantech/homebrew-core/actions/workflows/update.yml/dispatches \
+        -d '{"ref": "auto-update", "inputs": {"alan_version": "'"$alanversion"'"}}' \
```
Comment (dfellis) [inline .github/workflows/release.yml:154]: Has this been tested? I'm pretty sure bash single quotes do not allow `$var` substitution. I also don't know what's going on with the `"'"$...` part.

I think this needs to be something like:

```bash
-d "{\"ref\": \"auto-update\", \"inputs\": {\"alan_version\": \"$alanversion\"}}" \
```

File: .github/workflows/release.yml (hunk 2 above)
Comment (aguillenv) [reply .github/workflows/release.yml:143]: Ok, will remove the TODO

File: .github/workflows/release.yml (hunk 1 above)
Comment (aguillenv) [reply .github/workflows/release.yml:146]: Yes, it is bash partial matching

**Comment by dfellis** [comment]
I'm more concerned with this one; testing it is hard. But once it's figured out, we should be able to use it as a pattern on the Arch Linux repo; we push it to Github as well, add an action there to auto-update itself with the SHAs of the alan and anycloud binaries, and then push to Arch's git server (after uploading the account access as a secret in github, as well).

File: .github/workflows/release.yml (hunk 3 above)
Comment (aguillenv) [reply .github/workflows/release.yml:154]: Yes, I have tested both, release and prerelease and it works. This give me some war but how I have it it works

Comment (aguillenv) [reply .github/workflows/release.yml:154]: I have to update the `ref` to `main`, though

**Review by dfellis — APPROVED** [review]

File: .github/workflows/release.yml (hunk 3 above)
Comment (aguillenv) [reply .github/workflows/release.yml:154]: The `"'"$...` it is because it is a JSON, in that way I concatenate strings

**Review by cdmistman — APPROVED** [review]

File: .github/workflows/release.yml (hunk 3 above)
Comment (dfellis) [reply .github/workflows/release.yml]: Oh! That's clever!

**Review by depombo — APPROVED** [review]

**Pull request merged**